### ISSUE TITLE
[REQUEST] Co-constraint and OBX-2 verification

### ISSUE THREAD

**Issue by carolinerosin** [usnistgov/hl7-igamt#1280]
**Is your feature request related to a problem? Please describe.**
Verification must catch if an OBX-2 value in the co-constraint is not present in the value set associated with OBX-2.
Verification must catch if there is no value in OBX-2 in a co-constraint.

This can happen if the user builds a co-constraint and sets a value for OBX-2, and after that goes to delete the value in the value set or modifies the binding and picks a value set without that value.

**Describe the solution you'd like**
If there is value for OBX-2 in the co-constraint that does not come from HL70125, I don't think this would break validation, but the user can create a profile that always fails. I think this should be categorized as ERROR. Not sure if the user interface would allow that, but it should be checked because could happen in the XML (if user edits the XML and then uploads to GVT).

If there is no value for OBX-2 (see screenshots below), then this is FATAL

**Describe alternatives you've considered**
n/a

**Additional context**
Example
Constraint table (based on initial HL70125) : the user picks "CE" (valid at this point because it is in HL70125)
![image](https://user-images.githubusercontent.com/13591511/123103971-83bb9b00-d404-11eb-80c6-6feeab55f892.png)

Then user creates a flavor of HL70125 and removes "CE". He assigns that flavor to OBX-2
![image](https://user-images.githubusercontent.com/13591511/123104512-02b0d380-d405-11eb-82d9-081fbaa9128b.png)
![image](https://user-images.githubusercontent.com/13591511/123104578-13f9e000-d405-11eb-951e-efc66ce86fa1.png)

The co-constraint table looks like that
![image](https://user-images.githubusercontent.com/13591511/123106369-9505a700-d406-11eb-8b02-59394174286a.png)

Verification does not pick anything for the moment (probably because we haven't thought about it yet)

**Comment by Jungyubw** [comment]
@carolinerosin  For verification,

If no value in OBX-2 in Co-Constraint ==> Error
If the value in OBX-2 in Co-Constraint is missing ==> Fatal
If the value in OBX-2 in Co-Constraint is not matched  with any value of OBX-2's valueset ==> Fatal?

Am I right?

**Comment by carolinerosin** [comment]
yes, that is correct. We can ask for confirmation to Rob for the second one, but my approach is that FATAL is something that is broken and cannot be processed by GVT/validation.
in the second case, the validation would work, it is just that the user would create a profile that always fails.
